### PR TITLE
Add change origin to assist form control directive

### PR DESCRIPTION
**Pull request by SEAjamieD**
### :nut_and_bolt: Description: What code changed, and why?
On the data lifecycle page, users were prevented from selecting "days" as an option in Client Runs : label node as missing, if they hadn't already altered something on the page.

This branch adds the `resetOrigin` property to the formControl inputs, as well as updates them on save, so that the user can save as many times as they want and the formControl will remember their last known value.

### :chains: Related Resources
Fixes part 2 of: https://github.com/chef/automate/issues/3651

### :+1: Definition of Done
Users can change the values to whatever they like.  If the value is different than before, the 'Save Changes' button will activate.  If the value is the changed back to its previous value prior to saving changes, the 'Save Changes' button will deactivate.

### :athletic_shoe: How to Build and Test the Change
in hab studio `build components/automate-ui-devproxy && start_all_services`
cd into components/automate-ui and run `make serve`

navigate to https://a2-dev.test/settings/data-lifecycle
unless disabled user should be able to change all values as desired and 'Save Changes' button is activated when values are different than previously set.  If they move back to the previous value prior to saving. Save Changes button is disabled again.

One thing to note, I've found that with the Number inputs, if user uses the mouse or arrow keys to alter the numbers, the formControl Directive values are not respected.  This does not prevent the user from sending any values, but we wont see the full value of the form control directive.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
After:
Chef Automate (5)

### CODE REVIEW

File: components/automate-ui/src/app/pages/automate-settings/automate-settings.component.html [hunk 1]
```diff
@@ -141,10 +141,10 @@ <h2>Client Runs</h2>
               <input chefInput 
                 type="number" 
                 min="0" 
-                formControlName="threshold" 
+                formControlName="threshold" [resetOrigin]="shouldResetValues"
                 (keydown)="preventNegatives($event.key)" />
```
Comment (msorens) [inline components/automate-ui/src/app/pages/automate-settings/automate-settings.component.html:145]: Suggest rewriting some of the elements so they appear consistent.
1. This `<input>` element is the most egregious example--it puts one attribute per line. All the other ones are on a single line. 
2. Put the attributes in the same order. It just helps lessen the mental effort needed to read the file. For instance, here are the first two in the file where things are in a different order:
```
<input chefInput formControlName="threshold" [resetOrigin]="shouldResetValues" type="number" min="0" 
<input chefInput type="number" min="0" formControlName="threshold" [resetOrigin]="shouldResetValues"
```

File: components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts [hunk 2]
```diff
@@ -216,8 +219,11 @@ export class AutomateSettingsComponent implements OnInit, OnDestroy {
             this.saving = false;
           } else if (changeConfigurationSelector.status === 'loadingSuccess') {
             this.showSuccessNotification();
-            this.automateSettingsForm.markAsPristine();
             this.saving = false;
+            // After a successful Save, we flip shouldResetValues to true
+            // to update the resetOrigin values
```
Comment (msorens) [inline components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts:224]: Consider rewording for clarity (harping on my theme--this is more behavior, less implementation 😁 ):
```suggestion
            // After a successful save, trigger a notification to FormControlDirective
            // to consider the newly saved values as the new "original" values.
```

File: components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts [hunk 3]
```diff
@@ -375,6 +381,9 @@ export class AutomateSettingsComponent implements OnInit, OnDestroy {
           break;
       }
     });
+    // AfterInit, we need to apply this for resetOrigin to reset the values of formcontrolDirective
```
Comment (msorens) [inline components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts:384]:
```suggestion
    // After a successful load of initial values, trigger a notification 
    // to FormControlDirective to treat them as the "original" values.
```

**Comment by SEAjamieD** [comment]
> Code looks reasonable at first glance, but it is not quite working for me:
> ✔️ Upon load Save is disabled
> ✔️ Change a value and Save becomes enabled
> ❌ Change it back--Save does not disable for any number (this does work for "days", "hours", "minutes")
> ✔️ Make and save changes and Save becomes disabled

Thanks for the feedback!

For the number inputs - curious if you saw my comment above so i'm gonna post it here:

One thing to note, I've found that with the Number inputs, if user uses the mouse or arrow keys to alter the numbers, the formControl Directive values are not respected. This does not prevent the user from sending any values, but we wont see the full value of the form control directive.

Curious if these worked for you when using the number keys, as opposed to the mouse or arrow keys.  Seems like a weird behavior.

**Comment by msorens** [comment]
Ah, right you are @SEAjamieD; I missed your note. It does work when typing values. Very odd. So that does put us in a better state, at least. 👍

**Comment by SEAjamieD** [comment]
> Ah, right you are @SEAjamieD; I missed your note. It does work when typing values. Very odd. So that does put us in a better state, at least. 👍

This is peculiar behavior - I went a bit further and learned that this is current behavior in any form if a text input is of type `number`.  `formControlDirective` also does not establish original values if the user uses the arrow pad on their keyboard to make their number choice.  I tested this in a few places by switching the input type to `number` instead of text.  It doesn't appear we've caught this before because nearly all of our inputs are of type `text`.

While I think that it's important for an update to be made for this, I don't feel that it is worthy of holding up this branch for two reasons.

1.  This bug may currently be preventing users from reverting _back_ to a value if it happened to be a default value.
2.  This branch will no longer prevent a user from executing the save they wish.  It will only allow them to execute the save more than once.  I feel the allowing a second save is far better than preventing an initial save.

I propose opening a non-urgent bug ticket to handle the disabling of save button when the user is using a mouse or arrow pad.

**Pull request merged**